### DISCUSSION TITLE
Dreamcast VMU issues

### DISCUSSION THREAD

**Discussion by Tronicgg** [darthcloud/BlueRetro#947] [Q&A]
Having a strange issue with VMU emulation on v1.9.2. I do get a VMU to show up on Dreamcast bios when connecting player 1 controller. I "format" it using the DC bios and all works great both inside bios and in-game. I can save and delete and whatnot. Issue is everytime I turn the DC off and back on the VMU gets "corrupted" and I have to format it again, loosing everything. Any ideas on what might be going on? Tried pretty much everything on Advance config.

I have a simple internal install with hw1, no fancy controller autodetection. I used a standard 30pin ESP32.

**Comment by darthcloud** [comment]
Did you had this issue on older version?

**Reply by Tronicgg** [reply]
Issue was consistent with old and new firmware. I pretty much tried everything in terms of config –only VMU, VMU+Rumble, GamepadAlt, etc– but nothing helped. Ended up redoing the install with another ESP32 just to troubleshoot and issue has gone away. Very strange, because first installation was checked and working fine. Anyway, I have been testing for a week and could not replicate the issue, so I consider it solved.

**Comment by HornheaDD** [comment]
Sorry to chime in here, but I seem to be experiencing the same issue. This was happening on 1.9, and now 1.9.2. I was playing PSO and saved (just to save a game) then when I stopped playing and came back later just now, I went to try and use that VMU "disc" that has tons of saves on it, but it wouldnt let me select anything (like basically wouldnt let me hit start to continue).  Rebooted and then booted the BIOS, and the VMU was corrupted. 

It shows up with an X with a circle on it, and clicking on it only allows you to delete/format it.  Its happened two or three times.  I do have my "prototype" that still plugs into the controller slot. It was an older version (1.8.4 i think?), before the VMU implementation, anyway.  So I updated it as well, and the VMU showed up. Added a save file to it, power cycled, it worked ok. But to test, I powered down again, and it was corrupted, had to format.

I tried soft rebooting (I have a DCDigital in it) and that doesnt seem to kill the VMU, but a power cycle does. It is still happens with both of the ones I built.

**Comment by darthcloud** [comment]
Can you guys try to get the full logs from initial power on, saving something, power off, then power on again?

See:
https://github.com/darthcloud/BlueRetro/wiki/Getting-BlueRetro-debug-logs-via-Serial-port-Windows-10

**Reply by Tronicgg** [reply]
It now has been two weeks and I've had no issues. Power cycled the console many times. Waited days before booting up again. I cannot replicate the behaviour I was having with the old install. VMU works 100% of the time. I am still not sure what was going on, but I guess I have to blame my old install or maybe my old ESP32? Not sure.

**Comment by HornheaDD** [comment]
Ok, Im not certain I got the logs correctly, it didnt look like the report 'finished', it just kept going back and forth between these two entries:

(#) BT_HCI_EVT_LE_ADVERTISING_REPORT
(#)BT_HCI_EVT_INQUIRY_COMPLETE

But I was able to save them to a notepad file. I ended up reflashing both of my ESP32 boards with the 1.9.2 update file, and the one that was external (hooked up to a Dreamcast controller cable) was ok and never messed up the VMU even on several power down/power ups.  The internal one kept having the issue, even after a reflash.  Maybe the board was going bad? 

I tried having both the VMU and Rumble options, just the VMU, etc. Same thing. What I ended up doing was just taking out the internal one and installing the one I had on the controller cable in its place. Now it seems to be working just fine. I'll attach the two logs here.

[failure-internal.txt](https://github.com/darthcloud/BlueRetro/files/15084177/failure-internal.txt) (the one with the issue)
[success-external.txt](https://github.com/darthcloud/BlueRetro/files/15084178/success-external.txt) (the one that replaced it)

edit: thanks for the assistance :)

**Reply by darthcloud** [reply]
The memory card file look bad on first one you could try delete mc.bin from the web:
https://blueretro.io/files.html

**Reply by HornheaDD** [reply]
Interesting, I didnt know that could be done. Thats pretty cool. Well, by now its a moot point because I removed it from the Dreamcast (though I was still able to connect to the board itself by giving it USB power and do that). I'll keep that in mind to see if the new one I installed starts to do the same thing.

**Comment by Tronicgg** [comment]
> Interesting, I didnt know that could be done. Thats pretty cool. Well, by now its a moot point because I removed it from the Dreamcast (though I was still able to connect to the board itself by giving it USB power and do that). I'll keep that in mind to see if the new one I installed starts to do the same thing.

Me neither. A file manager is news to me. Will also keep this in mind if the issue manifests again in the future.

**Comment by raygan** [comment]
I'm not sure I can add much since I didn't get a log file, but I was having this issue on my AliExpress-purchased BlueRetro that I had flashed the 24.04 DC firmware on. Deleting the mc.bin file using the file manager did not solve the issue. Re-flashing the firmware (from the PC over USB) does seem to have solved it though, at least for now. If it recurs I'll try to get a full log. In the mean time, I'm just making frequent backups of my VMU file.

**Comment by davewongillies** [comment]
I've got a BlueRetro Core and am also seeing this issue. Tried deleting mc.bin and also have done a full factory reset and updated the firmware to 24.04 but it hasn't changed anything. Also it seems like my BR is now taking forever to pair up with any controllers. I'll try and get some logs.

**Comment by cecoates** [comment]
I have a BlueRetro Core and am seeing a similar issue. However I am never able to get the VMU recognized from the Dreamcast BIOS.

I can see it, choose the option to delete all and reformat it. But it remains as a big "X".

It happens on both 24_04 and 24_10. Deleting the mc.bin doesn't seem to help. I have also factory reset and updated the firmware (I was switching back and forth between 24.04 and 24.10 to see if that helped).

I did try flashing 1.9.2, but on that version my Xbox gamepad can't seem to connect to the BlueRetro Core.

I have generally tried this:

1. Factory reset
2. Flash firmware over USB
3. Set controller output 1 to use the memory card from https://blueretro.io/advance.html
4. Power cycle the BlueRetro, then try connecting to the DC.
5. Attempt to format the VMU via the Dreamcast BIOS.

Attaching logs from when I originally factory reset/flashed 24.10 (there's a new log whenever I power cycled and Putty disconnected). Including changing the Advanced settings to use the memory card/rumble, booting the Dreamcast and trying to erase the VMU so it can be used.

At some point in that process I also tried deleting the mc.bin and rebooting the BlueRetro. But each time even after the "Delete all (memory reset)" step in the Dreamcast BIOS, it remains a big "X".

[24.10-1.txt](https://github.com/user-attachments/files/17416399/24.10-1.txt)
[24.10-2.txt](https://github.com/user-attachments/files/17416400/24.10-2.txt)
[24.10-3.txt](https://github.com/user-attachments/files/17416401/24.10-3.txt)

Edit: This seems to only happen when using the Universal firmware. I flashed the Dreamcast specific one and the memory card worked immediately. Since the Core lets you use different cables, I was trying to leave it on Universal so it would autodetect the system.

**Reply by davewongillies** [reply]
I went through the same process as you. In the end I found that flashing my BlueRetro Core with the DC specific firmware (not the universal) made the VMU work on my Dreamcast, which isn't ideal as I'd like to use this on other systems. But that's less of an issue for me now as I bought a second one as I wasn't sure if it was because my adapter was faulty or not.
